pkg/commands/git_commands/existing_main_branches.go

```diff
@@ -0,0 +1,116 @@
+package git_commands
+
+import (
+	"strings"
+	"sync"
+
+	"github.com/jesseduffield/lazygit/pkg/commands/oscommands"
+	"github.com/jesseduffield/lazygit/pkg/utils"
+	"github.com/samber/lo"
+	"github.com/sasha-s/go-deadlock"
+)
+
+type ExistingMainBranches struct {
+	configuredMainBranches []string
```

Comment:
I like `MainBranches`; renamed in e702411f0486, and also added a few comments.

This one was an interesting exercise for ctrl-f. Just hitting ctrl-f didn't work of course, because the changes needed to go into 5 different commits. I tried to run git-absorb on all changes at once, and it did manage to create 5 fixup commits automatically. However, some of them were wrong, i.e. after squashing the fixups, some commits didn't compile. Creating the fixups manually one by one using ctrl-f was more work because I had to stage the changes that belong together, but the result was then correct.